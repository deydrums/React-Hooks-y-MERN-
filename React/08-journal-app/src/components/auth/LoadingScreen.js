import React from 'react'

export const LoadingScreen = () => {
    return (
        <div>
            <div className="auth__main">
                <div className="auth__loading">
                <div class="page-loader">
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
                </div>
            </div>
        </div>
    )
}
