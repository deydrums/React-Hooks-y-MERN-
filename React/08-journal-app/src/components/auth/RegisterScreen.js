import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {Link} from 'react-router-dom';
import validator from 'validator';
import { removeError, setError } from '../../actions/ui';
import { useForm } from '../../hooks/useForm';

export const RegisterScreen = () => {

    const dispatch = useDispatch();
    const {msgError} = useSelector(state =>state.ui);

    const [formValues, handleInputChange, reset] = useForm({
        name: 'admin',
        email: 'admin@admin.com',
        password: '123456',
        password2: '123456'
    });

    const {name, email, password, password2} = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        if (isFormValid()){
            console.log('Formulario Correcto')
        }
    };

    const isFormValid = () =>{
        if(name.trim().length ===0){
            dispatch(setError('Nombre es requerido'));
            return false;
        }else if(!validator.isEmail(email)){
            dispatch(setError('Email invalido'));
            return false;
        }else if (password !== password2 || password.length < 5){
            dispatch(setError('Contraseñas no coinciden (deben ser mayor a 4 caracteres)'));
            return false;
        }
        dispatch(removeError());
        return true;
    };

    return (
        <>
            <h3 className="auth__title">Registrarse</h3>
            {            
                msgError &&
                <div className="auth__alert-error"> 
                    {msgError}
                </div>
            }            
            <form onSubmit = {handleRegister}>
                <input
                    type="text"
                    placeholder="Nombre..."
                    name = "name"
                    className="auth__input"
                    autoComplete="off"
                    onChange = {handleInputChange}
                    value = {name}
                />
                <input
                    type="text"
                    placeholder="Email..."
                    name = "email"
                    className="auth__input"
                    autoComplete="off"
                    onChange = {handleInputChange}
                    value = {email}
                />
                <input
                    type="password"
                    placeholder="Contraseña..."
                    name = "password"
                    className="auth__input"
                    onChange = {handleInputChange}
                    value = {password}
                />
                <input
                    type="password"
                    placeholder="Confirmar contraseña..."
                    name = "password2"
                    className="auth__input"
                    onChange = {handleInputChange}
                    value = {password2}
                />
                <button
                    type="submit"
                    className = "btn btn-primary btn-block mb-3"
                >
                    Registrarse
                </button>
              
                <Link 
                    to="/auth/login"
                    className="link"
                >
                    Logueate aquí
                </Link>
            </form>
        </>
    )
}
