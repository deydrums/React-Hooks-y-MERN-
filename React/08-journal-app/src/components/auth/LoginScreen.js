import React from 'react';
import {Link} from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { useDispatch } from 'react-redux';
import { login } from '../../actions/auth';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [formValues, handleInputChange, reset] = useForm({
        email: 'admin@admin.com',
        password: '123456'
    });

    const {email, password} = formValues;

    const handleLogin = (e) => {
        e.preventDefault();
        dispatch(login(12345,'David'));
    }

    return (
        <>
            <h3 className="auth__title">Iniciar sesión</h3>
            <form onSubmit={handleLogin}>
                <input
                    type="text"
                    placeholder="Email..."
                    name = "email"
                    className="auth__input"
                    autoComplete="off"
                    value = {email}
                    onChange={handleInputChange}
                />
                <input
                    type="password"
                    placeholder="Password..."
                    name = "password"
                    className="auth__input"
                    value = {password}
                    onChange={handleInputChange}
                />
                <button
                    type="submit"
                    className = "btn btn-primary btn-block"
                >
                    Entrar
                </button>
                
                <div className="auth__social-networks">
                    <p>Iniciar con redes sociales</p>
                    <div 
                        className="google-btn"
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Entrar con Google</b>
                        </p>
                    </div>
                </div>
                <Link 
                    to="/auth/register"
                    className="link"
                >
                    Registrarse aquí
                </Link>
            </form>
        </>
    )
}
