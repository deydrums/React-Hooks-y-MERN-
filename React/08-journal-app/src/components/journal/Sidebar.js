import React from 'react';
import { JournalEntries } from './JournalEntries';
import { useDispatch } from 'react-redux';
import { startLogout } from '../../actions/auth';

export const Sidebar = () => {

    const dispatch = useDispatch();

    const handleLogout = () => {
       dispatch(startLogout());
    }
    return (
        <nav className = "journal__sidebar">
            <div className="journal__sidebar-navbar">
                <h3 className = "mt-5">
                    <i className="fa fa-moon"></i>
                    <span> David</span>
                </h3>

                <button 
                    className="btn"
                    onClick = {handleLogout}
                >
                    Cerrar sesiÃ³n
                </button>
            </div>
            <div className="journal__new-entry">
                <i className="far fa-calendar-plus fa-5x"></i>
                <p className = "mt-5">Nueva entrada</p>
            </div>
            <JournalEntries/>
        </nav>
    )
}
