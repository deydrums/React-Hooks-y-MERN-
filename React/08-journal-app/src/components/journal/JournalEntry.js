import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';
import { activeNote } from '../../actions/notes';

export const JournalEntry = ({id, date, title, body, url}) => {
    const dispatch = useDispatch(); 
    moment.updateLocale('es', {
        months: 'Enero_Febrero_Marzo_Abril_Mayo_Junio_Julio_Agosto_Septiembre_Octubre_Noviembre_Diciembre'.split('_'),
        monthsShort: 'Enero._Feb._Mar_Abr._May_Jun_Jul._Ago_Sept._Oct._Nov._Dec.'.split('_'),
        weekdays: 'Domingo_Lunes_Martes_Miercoles_Jueves_Viernes_Sabado'.split('_'),
        weekdaysShort: 'Dom._Lun._Mar._Mier._Jue._Vier._Sab.'.split('_'),
        weekdaysMin: 'Do_Lu_Ma_Mi_Ju_Vi_Sa'.split('_')
      })
    const noteDate = moment(date);
    const handleEntryClick = () => {
        dispatch(activeNote(id,{
            date,title,body,url
        }));
    }

    return (
        <div className="journal__entry pointer animate__animated animate__fadeIn" onClick={handleEntryClick}>
            {
                url ?
                <div 
                    className="journal__entry-picture"
                    style = {{
                        backgroundSize: 'cover',
                        backgroundImage: `url(${url})`
                    }}
                >
                </div>
                :
                <div 
                className="journal__entry-picture"
                style = {{
                    backgroundSize: 'cover',
                    backgroundColor: 'rgba(102, 250, 193, 0.699)'
                }}
                >
                </div>
            }
            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    {title}
                </p>
                <p className="journal__entry-content">
                    {body}
                </p>
            </div>
            <div className="journal__entry-date-box">
                <span>{noteDate.format('dddd')}</span>
                <h4>{noteDate.format('Do MMM')}</h4>
            </div>
        </div>
    )
}
