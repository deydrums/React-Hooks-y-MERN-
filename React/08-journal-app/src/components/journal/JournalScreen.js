import React from 'react';
import { NoteScreen } from '../notes/NoteScreen';
// import { NothingSelected } from './NothingSelected';
import { Sidebar } from './Sidebar';

export const JournalScreen = () => {
    const btnmoreClick = () => { 
        const q = document.querySelector('.journal__sidebar');
        q.classList.toggle('journal__active');
    }
    const btnmoreOutClick = () => { 
        const q = document.querySelector('.journal__sidebar');
        q.classList.remove('journal__active');
    }
    
    return (
        <div className = "journal__main-content animate__animated animate__fadeIn" >
            <Sidebar/>
            <main onClick = {btnmoreOutClick}>
                {/* <NothingSelected/> */}
                <NoteScreen />
            </main>
            <a 
                className = "journal__btnmore"
                onClick={btnmoreClick}
            >
                <i className="far fa-calendar-plus fa-2x"></i>
            </a>
        </div>
    )
}
