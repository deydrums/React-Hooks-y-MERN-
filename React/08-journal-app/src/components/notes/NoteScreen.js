import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    return (
        <div className = "notes__main-content">
            <NotesAppBar />
            <div className = "notes__content">
                    <input
                        type = "text"
                        placeholder = "TÃ­tulo..."
                        className = "notes__title-input"
                    />
                    <textarea
                        placeholder = "Que paso hoy?"
                        className = "notes__textarea"
                    >

                    </textarea>

                    <div className = "notes__image">
                        <img 
                            src = "https://64.media.tumblr.com/b38f6615b179cff391fc31456895f6ab/9d110a7eb7405c5c-6f/s2048x3072/f8905f13c648107d1f218b9902f588dfd4ddcb6e.jpg"
                            alt = "image"
                        />
                    </div>

            </div>
        </div>
    )
}
