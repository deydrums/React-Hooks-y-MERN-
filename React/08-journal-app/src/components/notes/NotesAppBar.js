import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {
    const dispatch = useDispatch();
    const {active} = useSelector(state => state.notes);
    const handleSave = () => {
        dispatch(startSaveNote(active));
    }

    const handlePictureUpload = () =>{
        document.querySelector('#fileSelector').click();
    }

    const handleFileChange = (e) => {
        const file = e.target.files;
        if(file){
            dispatch(startUploading(file));
        }
    }

    return (
        <div className="notes__appbar">
            <span>28 de agosto 2021</span>
            <input
                type="file"
                style = {{display:'none'}}
                onChange={handleFileChange}
                id = "fileSelector"
                name = "file"
            />

            <div>
                <button className="btn" onClick = {handlePictureUpload}>
                    Fotograf√≠a
                </button>
                <button className="btn" onClick={handleSave}>
                    Guardar
                </button>
            </div>
        </div>
    )
}
