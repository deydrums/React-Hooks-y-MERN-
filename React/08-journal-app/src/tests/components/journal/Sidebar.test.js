import React from 'react';
import { mount } from 'enzyme';
import { Sidebar } from '../../../components/journal/Sidebar';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { startLogout } from '../../../actions/auth';
import { startNewNote } from '../../../actions/notes';

jest.mock( '../../../actions/auth', () => ({
    startLogout: jest.fn(),
}));
jest.mock( '../../../actions/notes', () => ({
    startNewNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {},
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: 'ABC',
        },
        notes: []
    }
};
const store = mockStore(initState);
store.dispatch = jest.fn();
describe('Pruebas en Sidebar', () => {

    const wrapper = mount(
        <Provider store={store}>
            <Sidebar/>
        </Provider>
    );

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de llamar el startLogout', () => {
        wrapper.find('button').prop('onClick')();
        expect(startLogout).toHaveBeenCalledTimes(1);

    });

    test('Debe de llamar el startNewNote', () => {
        wrapper.find('.journal__new-entry').prop('onClick')();
        expect(startNewNote).toHaveBeenCalledTimes(1);
    });
    
});
