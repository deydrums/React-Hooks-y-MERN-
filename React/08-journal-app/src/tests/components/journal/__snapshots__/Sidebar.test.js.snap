// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Sidebar Debe de mostrarse correctamente 1`] = `
<nav
  className="journal__sidebar"
>
  <div
    className="journal__sidebar-btnlogout"
  >
    <button
      className="btn"
      onClick={[Function]}
    >
      Cerrar sesi√≥n
    </button>
  </div>
  <div
    className="journal__sidebar-navbar"
  >
    <i
      className="fa fa-moon"
    />
    <h3>
      <span>
         
      </span>
    </h3>
  </div>
  <div
    className="journal__new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      Nueva entrada
    </p>
  </div>
  <div
    className="journal__entries "
  />
</nav>
`;
